// @fluentui/react-northstar-fela-renderer/dist/dts/src/felaExpandCssShorthandsPlugin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.felaExpandCssShorthandsPlugin = void 0;
// @ts-ignore
var inline_style_expand_shorthand_1 = require("inline-style-expand-shorthand");
// https://jsperf.com/array-indexof-vs-object-key-lookup2/12
var handledCssProps = {
    // 'font', Oops, is not supported by inline-style-expand-shorthand
    padding: true,
    margin: true,
    border: true,
    borderWidth: true,
    borderStyle: true,
    borderColor: true,
    borderTop: true,
    borderRight: true,
    borderBottom: true,
    borderLeft: true,
    borderRadius: true,
    background: true,
    outline: true,
};
var felaExpandCssShorthandsPlugin = function (styles) {
    return Object.keys(styles).reduce(function (
    // Without casting to any TSC gives "Expression produces a union type that is too complex to represent" error
    acc, cssPropertyName) {
        var cssPropertyValue = styles[cssPropertyName];
        if (cssPropertyValue === null || typeof cssPropertyValue === 'undefined') {
            acc[cssPropertyName] = cssPropertyValue;
        }
        else if (handledCssProps[cssPropertyName]) {
            var expandedProps = inline_style_expand_shorthand_1.expandProperty(cssPropertyName, cssPropertyValue);
            if (expandedProps) {
                Object.assign(acc, expandedProps);
                return acc;
            }
            acc[cssPropertyName] = cssPropertyValue;
        }
        else if (Array.isArray(cssPropertyValue)) {
            acc[cssPropertyName] = cssPropertyValue;
        }
        else if (typeof cssPropertyValue === 'object') {
            acc[cssPropertyName] = exports.felaExpandCssShorthandsPlugin(cssPropertyValue);
        }
        else {
            acc[cssPropertyName] = cssPropertyValue;
        }
        return acc;
    }, {});
};
exports.felaExpandCssShorthandsPlugin = felaExpandCssShorthandsPlugin;
