// @fluentui/react-bindings/dist/dts/src/hooks/useStyles.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useStyles = void 0;
var React = require("react");
var classnames_1 = require("classnames");
var context_1 = require("../context");
var getStyles_1 = require("../styles/getStyles");
var useStyles = function (displayName, options) {
    var _a, _b, _c;
    var context = context_1.useFluentContext();
    // Stores debug information for component.
    var debug = React.useRef({ fluentUIDebug: null });
    if (options.unstyled) {
        var componentClassName = ((_a = options.composeOptions) === null || _a === void 0 ? void 0 : _a.className) || options.className;
        var classNameProp = (_b = options.mapPropsToInlineStyles) === null || _b === void 0 ? void 0 : _b.call(options).className;
        return { classes: { root: classnames_1.default(componentClassName, classNameProp) }, styles: {} };
    }
    var _d = options.className, className = _d === void 0 ? process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉' : _d, composeOptions = options.composeOptions, _e = options.mapPropsToStyles, mapPropsToStyles = _e === void 0 ? function () { return ({}); } : _e, _f = options.mapPropsToInlineStyles, mapPropsToInlineStyles = _f === void 0 ? function () { return ({}); } : _f, _g = options.unstable_props, unstable_props = _g === void 0 ? {} : _g, _h = options.rtl, rtl = _h === void 0 ? false : _h;
    var componentStylesProps = mapPropsToStyles();
    // `composeProps` should include all props including stylesProps as they can contain state
    var composeProps = __assign(__assign({}, unstable_props), componentStylesProps);
    var composeStylesProps = (_c = composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.mapPropsToStylesPropsChain) === null || _c === void 0 ? void 0 : _c.reduce(function (acc, fn) { return (__assign(__assign({}, acc), fn(composeProps))); }, {});
    var _j = getStyles_1.getStyles({
        // Input values
        allDisplayNames: (composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.displayNames) || [displayName],
        className: (composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.className) || className,
        primaryDisplayName: (composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.displayName) || displayName,
        componentProps: __assign(__assign({}, componentStylesProps), composeStylesProps),
        inlineStylesProps: mapPropsToInlineStyles(),
        // Context values
        disableAnimations: context.disableAnimations,
        renderer: context.renderer,
        rtl: rtl,
        saveDebug: function (fluentUIDebug) { return (debug.current = { fluentUIDebug: fluentUIDebug }); },
        theme: context.theme,
        performance: context.performance,
        telemetry: context.telemetry,
    }), classes = _j.classes, resolvedStyles = _j.styles;
    return { classes: classes, styles: resolvedStyles };
};
exports.useStyles = useStyles;
