// @fluentui/accessibility/dist/dts/src/behaviors/Pill/pillBehavior.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.pillBehavior = void 0;
var keyboard_key_1 = require("../../keyboard-key");
var attributes_1 = require("../../attributes");
var pillBehavior = function (p) {
    var _a;
    return ({
        attributes: {
            root: __assign((_a = { role: p.actionable || p.selectable ? 'button' : 'none', tabIndex: p.actionable || p.selectable ? 0 : -1 }, _a[attributes_1.IS_FOCUSABLE_ATTRIBUTE] = p.actionable || p.selectable || p.role === 'option', _a), (p.selectable && {
                'aria-selected': p.selected,
            })),
        },
        keyActions: {
            root: __assign(__assign({}, (p.dismissible && {
                performDismiss: {
                    keyCombinations: [{ keyCode: keyboard_key_1.keyboardKey.Delete }, { keyCode: keyboard_key_1.keyboardKey.Backspace }],
                },
            })), ((p.selectable || p.actionable) && {
                performClick: {
                    keyCombinations: [{ keyCode: keyboard_key_1.keyboardKey.Enter }, { keyCode: keyboard_key_1.SpacebarKey }],
                },
            })),
        },
    });
};
exports.pillBehavior = pillBehavior;
